main.go

```diff
@@ -214,17 +209,16 @@ func setupControllers(ctx context.Context, mgr ctrl.Manager, cCache *cache.Cache
 		os.Exit(1)
 	}
 
-	crds := findCustomResources(ctx, mgr)
-
 	opts := []jobframework.Option{
 		jobframework.WithManageJobsWithoutQueueName(manageJobsWithoutQueueName),
 		jobframework.WithWaitForPodsReady(waitForPodsReady(cfg)),
 		jobframework.WithKubeServerVersion(serverVersionFetcher),
 	}
 	err := jobframework.ForEachIntegration(func(name string, cb jobframework.IntegrationCallbacks) error {
 		log := setupLog.WithValues("jobFrameworkName", name)
-		if isFrameworkEnabled(cfg, name) && crds.Has(name) {
-			if err := cb.NewReconciler(
+		gvk := cb.JobType.GetObjectKind().GroupVersionKind()
+		if _, err := mgr.GetRESTMapper().RESTMapping(gvk.GroupKind(), gvk.Version); err != nil && isFrameworkEnabled(cfg, name) {
```

Comment:
`isFrameworkEnabled(cfg, name) && crds.Has(name)` was the happy path. So IIUC it should correspond to no error (err= nil). 

Also, make sure the error we get is not lost. At least log it, but probably we should propagate it up. 

I would also do any checks only if the `isFrameworkEnabled` responds to true. Otherwise we risk logging an error or propagating up for frameworks which are disabled.

what is the `err` if there is no mapping, is it NotFound? Maybe there are 3 cases:
- no error -> happy path
- NotFound error- > path for !crds.Has(name)
- other generic error -> propagate the error

Reply:
> isFrameworkEnabled(cfg, name) && crds.Has(name) was the happy path. So IIUC it should correspond to no error (err= nil).

Yes, that's right. It's my bad :(

Reply:
> what is the err if there is no mapping,

It's expected `meta.IsNoMatchError`.

Reply:
Since the controller-runtime v0.15, the returned error seems to be changed with `discovery.ErrGroupDiscoveryFailed `.

And the changing seems the unintended breaking changes:
https://github.com/kubernetes-sigs/controller-runtime/pull/2425